external/elasticsearch/src/main/java/com/digitalpebble/stormcrawler/elasticsearch/persistence/AggregationSpout.java

```diff
@@ -255,8 +256,9 @@ public void onResponse(SearchResponse response) {
                 if (hitsForThisBucket == 1 && !iterator.hasNext()) {
                     String strDate = (String) keyValues.get("nextFetchDate");
                     try {
-                        mostRecentDateFound = formatter.parse(strDate);
-                    } catch (ParseException e) {
+                        LocalDateTime ldt = LocalDateTime.parse(strDate, formatter);
+                        mostRecentDateFound = Date.from(ldt.atZone(ZoneId.systemDefault()).toInstant());
```

Comment:
using `ZoneId.systemDefault()`, since when we are creating the calendar using Calendar.getInstance() few lines bellow, we are system default too

Reply:
Do we need to go via a Date?

Instant instant = ldt.toInstant(ZoneOffset.UTC);

with the default instead of ZOneOffset?

Or even better, if the ISO format matches the pattern we have above, wouldn't 

https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html#parse-java.lang.CharSequence-

do the trick?

Reply:
I will try, I like your suggestion...